executor/src/witgen/mod.rs

```diff
@@ -121,44 +132,52 @@ where
         let mut row_values = None;
         if let Some(period) = looping_period {
             let values = &rows[rows.len() - period];
-            if generator.propose_next_row(row, values) {
+            if generator.propose_next_row(row, values, &mut mutable_state) {
                 row_values = Some(values.clone());
             } else {
                 log::info!("Using loop failed. Trying to generate regularly again.");
                 looping_period = None;
             }
         }
         if row_values.is_none() {
-            row_values = Some(generator.compute_next_row(row));
+            row_values = Some(generator.compute_next_row(row, &mut mutable_state));
         };
 
         rows.push(row_values.unwrap());
     }
 
     // Transpose the rows
-    let mut columns = fixed.witness_map_with(vec![]);
+    let mut columns = base_witnesses
+        .iter()
+        .map(|c| (fixed.column_name(c).to_string(), vec![]))
+        .collect::<BTreeMap<_, _>>();
     for row in rows.into_iter() {
-        for (col_index, value) in row.into_iter() {
-            columns[&col_index].push(value);
+        for poly_id in &base_witnesses {
+            columns
+                .get_mut(fixed.column_name(poly_id))
```

Comment:
Didn't use `.entry(fixed.column_name(c).to_string()).or_insert_with()` so we don't have to call `to_string()` for every cell.

Reply:
I think even calling `column_name` is way too expensive here.
Did you check performance?
We could use a map from poly id and only replace it by the name as the very last step.

Reply:
True, it got slower (2s -> 12s for the transposing on the Keccak-via-RISC-V example), nice catch! Will fix that.

Just curious, you think it's the call to `column_name`, which is just return a `&str` without building any strings? I'm guessing it's more that comparison between strings is more expensive (needed for looking up the column vector for every cell).

Reply:
Yes sorry, of course it is the comparisons and not looking up the name.